crates/oxc_linter/src/rules/jest/prefer_expect_resolves.rs

```diff
@@ -0,0 +1,267 @@
+    /// });
+    /// it('is true', async () => {
+    ///     const myPromise = Promise.resolve(true);
+    ///
+    ///     await expect(myPromise).resolves.toBe(true);
+    /// });
+    ///
+    /// it('errors', async () => {
+    ///     await expect(Promise.reject(new Error('oh noes!'))).rejects.toThrowError(
+    ///         'oh noes!',
+    ///     );
+    /// });
+    ///
+    /// // invalid
+    /// it('passes', async () => {
+    ///     expect(await someValue()).toBe(true);
+    /// });
+    /// it('is true', async () => {
+    ///     const myPromise = Promise.resolve(true);
+    ///     expect(await myPromise).toBe(true);
+    /// });
+    /// ```
+    PreferExpectResolves,
+    style,
+);
+
+impl Rule for PreferExpectResolves {
+    fn run_once(&self, ctx: &LintContext) {
+        for possible_jest_node in &collect_possible_jest_call_node(ctx) {
+            Self::run(possible_jest_node, ctx);
+        }
+    }
+}
+
+impl PreferExpectResolves {
+    fn run<'a>(possible_jest_node: &PossibleJestNode<'a, '_>, ctx: &LintContext<'a>) {
+        let node = possible_jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Some(expect_fn_call) = parse_expect_jest_fn_call(call_expr, possible_jest_node, ctx)
+        else {
+            return;
+        };
+        let Some(parent) = expect_fn_call.head.parent else {
+            return;
+        };
+
+        if !parent.is_call_expression() {
+            return;
+        }
+
+        let Some(Expression::CallExpression(call_expr)) = expect_fn_call.head.parent else {
+            return;
+        };
+        let Some(argument) = call_expr.arguments.first() else {
+            return;
+        };
+        let Argument::Expression(Expression::AwaitExpression(await_expr)) = argument else {
+            return;
+        };
+
+        ctx.diagnostic_with_fix(ExpectResolves(await_expr.span), || {
+            let content = Self::build_code(call_expr, ctx);
+            Fix::new(content, call_expr.span)
+        });
+    }
+
+    fn build_code<'a>(call_expr: &CallExpression<'a>, ctx: &LintContext<'a>) -> String {
+        let mut formatter = ctx.codegen();
+        let ident = call_expr.callee.get_identifier_reference().unwrap();
+        let first = call_expr.arguments.first().unwrap();
+        let Argument::Expression(Expression::AwaitExpression(await_expr)) = first else {
+            return formatter.into_source_text();
+        };
+
+        let offset = match &await_expr.argument {
+            Expression::CallExpression(call_expr) => call_expr.span.start - ident.span.end,
+            Expression::Identifier(promise_ident) => promise_ident.span.start - ident.span.end,
+            _ => 0,
+        };
+
+        let arg_span = Span::new(
+            call_expr.span.start + (ident.span.end - ident.span.start) + offset,
+            call_expr.span.end,
+        );
```

Comment:
Do you mind adding some comment here to show the `offset` and `arg_span` :)

Reply:
There is no `span` in `await_expr.argument`, but it' not enough. because the keyword of `await` in expect need to be removed as well.